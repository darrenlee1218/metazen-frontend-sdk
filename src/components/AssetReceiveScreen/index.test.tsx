import { constants } from 'ethers';
import React, { ComponentProps } from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';

import { AssetReceiveScreen, ReceiveVariant } from '.';
import { TestProviders } from '@utils/testing/TestProviders';
import { TokenStandard } from '@gryfyn-types/data-transfer-objects/Token';

afterEach(cleanup);

jest.mock('react-router-dom');

const defaultProps: ComponentProps<typeof AssetReceiveScreen> = {
  variant: ReceiveVariant.Nft,
  recipientAddress: constants.AddressZero,
  displayName: 'ASSET',
  chainName: 'Polygon (Mumbai)',
  networkIconUrl: 'networkIcon.png',
  onClose: () => {},
  onBackPress: () => {},
};

it('should return to the homepage after clicking on close', () => {
  const mockedOnClose = jest.fn();
  const props = { ...defaultProps, onClose: mockedOnClose };
  render(<AssetReceiveScreen {...props} />, { wrapper: TestProviders });
  fireEvent.click(screen.getByText('Close'));
  expect(mockedOnClose).toHaveBeenCalled();
});

it('matches snapshot when variant is ReceiveVariant.Nft', () => {
  const { container } = render(<AssetReceiveScreen {...defaultProps} />, { wrapper: TestProviders });
  expect(container).toMatchSnapshot();
});

it('matches snapshot when variant is ReceiveVariant.Token', () => {
  const props = {
    ...defaultProps,
    variant: ReceiveVariant.Token,
    balance: {
      token: {
        ticker: 'ASSET',
        standard: TokenStandard.ERC20,
        display: { svgIconUrl: 'icon.png' },
        decimal: 18,
        displayPrecision: 4,
        custodyPrecision: 8,
      },
      balanceInAssetDecimals: '10000000000000000000',
      price: '1',
    } as unknown as any,
  };
  const { container } = render(<AssetReceiveScreen {...props} />, { wrapper: TestProviders });
  expect(container).toMatchSnapshot();
});
