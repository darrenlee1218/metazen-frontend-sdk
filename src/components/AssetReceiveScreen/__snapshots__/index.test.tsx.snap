// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when variant is ReceiveVariant.Nft 1`] = `
<div>
  <div
    class="MuiBox-root css-tfe0o3"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1rgr0y-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xtuuzo-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-16vochq-MuiTypography-root"
    >
      Receive ASSET
    </h2>
    <div
      class="MuiBox-root css-8h2w1h"
    />
  </div>
  <div
    class="MuiBox-root css-dodud4"
  >
    <div
      class="MuiBox-root css-1010uug"
    >
      <svg
        height="180"
        shape-rendering="crispEdges"
        viewBox="0 0 37 37"
        width="180"
      >
        <path
          d="M0,0 h37v37H0z"
          fill="#111619"
        />
        <path
          d="M0 0h7v1H0zM8 0h2v1H8zM13 0h4v1H13zM18 0h2v1H18zM21 0h1v1H21zM27 0h2v1H27zM30,0 h7v1H30zM0 1h1v1H0zM6 1h1v1H6zM10 1h6v1H10zM17 1h4v1H17zM22 1h4v1H22zM27 1h1v1H27zM30 1h1v1H30zM36,1 h1v1H36zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM10 2h3v1H10zM14 2h1v1H14zM16 2h1v1H16zM22 2h1v1H22zM24 2h1v1H24zM26 2h1v1H26zM28 2h1v1H28zM30 2h1v1H30zM32 2h3v1H32zM36,2 h1v1H36zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM10 3h4v1H10zM16 3h2v1H16zM20 3h2v1H20zM27 3h2v1H27zM30 3h1v1H30zM32 3h3v1H32zM36,3 h1v1H36zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h1v1H9zM11 4h3v1H11zM15 4h4v1H15zM20 4h2v1H20zM24 4h1v1H24zM27 4h2v1H27zM30 4h1v1H30zM32 4h3v1H32zM36,4 h1v1H36zM0 5h1v1H0zM6 5h1v1H6zM9 5h2v1H9zM13 5h2v1H13zM18 5h2v1H18zM21 5h1v1H21zM23 5h1v1H23zM25 5h1v1H25zM27 5h1v1H27zM30 5h1v1H30zM36,5 h1v1H36zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22 6h1v1H22zM24 6h1v1H24zM26 6h1v1H26zM28 6h1v1H28zM30,6 h7v1H30zM9 7h4v1H9zM14 7h3v1H14zM22 7h2v1H22zM25 7h1v1H25zM27 7h1v1H27zM2 8h1v1H2zM4 8h3v1H4zM8 8h1v1H8zM10 8h1v1H10zM16 8h1v1H16zM18 8h1v1H18zM21 8h2v1H21zM24 8h3v1H24zM29 8h1v1H29zM33 8h1v1H33zM36,8 h1v1H36zM2 9h1v1H2zM7 9h2v1H7zM10 9h3v1H10zM14 9h1v1H14zM16 9h1v1H16zM19 9h1v1H19zM23 9h4v1H23zM30 9h2v1H30zM33 9h1v1H33zM36,9 h1v1H36zM0 10h3v1H0zM6 10h4v1H6zM13 10h1v1H13zM15 10h1v1H15zM19 10h1v1H19zM21 10h1v1H21zM23 10h2v1H23zM26 10h1v1H26zM28 10h1v1H28zM30 10h1v1H30zM32 10h1v1H32zM34 10h1v1H34zM36,10 h1v1H36zM1 11h2v1H1zM5 11h1v1H5zM11 11h4v1H11zM16 11h1v1H16zM18 11h1v1H18zM20 11h1v1H20zM24 11h2v1H24zM27 11h1v1H27zM29 11h1v1H29zM31 11h1v1H31zM33 11h1v1H33zM35 11h1v1H35zM4 12h7v1H4zM12 12h1v1H12zM15 12h2v1H15zM18 12h1v1H18zM20 12h1v1H20zM25 12h2v1H25zM30 12h2v1H30zM33 12h1v1H33zM36,12 h1v1H36zM0 13h1v1H0zM2 13h1v1H2zM5 13h1v1H5zM7 13h1v1H7zM9 13h4v1H9zM14 13h2v1H14zM22 13h5v1H22zM30 13h2v1H30zM33 13h1v1H33zM36,13 h1v1H36zM1 14h3v1H1zM6 14h2v1H6zM9 14h1v1H9zM11 14h2v1H11zM16 14h1v1H16zM19 14h1v1H19zM21 14h3v1H21zM26 14h1v1H26zM28 14h1v1H28zM30 14h1v1H30zM32 14h1v1H32zM34 14h1v1H34zM36,14 h1v1H36zM1 15h1v1H1zM3 15h2v1H3zM7 15h2v1H7zM14 15h1v1H14zM17 15h1v1H17zM19 15h1v1H19zM21 15h3v1H21zM25 15h1v1H25zM27 15h1v1H27zM29 15h1v1H29zM31 15h1v1H31zM35 15h1v1H35zM0 16h3v1H0zM6 16h1v1H6zM8 16h2v1H8zM13 16h7v1H13zM21 16h1v1H21zM23 16h1v1H23zM25 16h2v1H25zM30 16h2v1H30zM33 16h1v1H33zM36,16 h1v1H36zM0 17h3v1H0zM4 17h2v1H4zM9 17h1v1H9zM12 17h1v1H12zM16 17h1v1H16zM18 17h1v1H18zM21 17h6v1H21zM30 17h2v1H30zM33 17h1v1H33zM36,17 h1v1H36zM0 18h1v1H0zM4 18h4v1H4zM10 18h3v1H10zM14 18h1v1H14zM16 18h4v1H16zM21 18h1v1H21zM24 18h1v1H24zM26 18h1v1H26zM28 18h1v1H28zM30 18h1v1H30zM32 18h1v1H32zM34 18h1v1H34zM36,18 h1v1H36zM0 19h2v1H0zM3 19h2v1H3zM8 19h1v1H8zM12 19h1v1H12zM15 19h1v1H15zM23 19h3v1H23zM27 19h1v1H27zM29 19h1v1H29zM31 19h1v1H31zM33 19h1v1H33zM35,19 h2v1H35zM1 20h3v1H1zM5 20h2v1H5zM10 20h2v1H10zM14 20h1v1H14zM17 20h1v1H17zM20 20h1v1H20zM24 20h3v1H24zM30 20h2v1H30zM33 20h1v1H33zM2 21h1v1H2zM8 21h2v1H8zM11 21h1v1H11zM16 21h5v1H16zM22 21h1v1H22zM25 21h2v1H25zM30 21h2v1H30zM33 21h1v1H33zM36,21 h1v1H36zM0 22h2v1H0zM3 22h1v1H3zM5 22h5v1H5zM11 22h1v1H11zM13 22h1v1H13zM16 22h1v1H16zM18 22h4v1H18zM23 22h2v1H23zM26 22h1v1H26zM28 22h1v1H28zM30 22h1v1H30zM32 22h1v1H32zM34 22h1v1H34zM36,22 h1v1H36zM0 23h4v1H0zM7 23h2v1H7zM12 23h2v1H12zM16 23h2v1H16zM20 23h1v1H20zM24 23h2v1H24zM27 23h1v1H27zM29 23h1v1H29zM31 23h1v1H31zM33 23h1v1H33zM35 23h1v1H35zM0 24h2v1H0zM4 24h6v1H4zM11 24h1v1H11zM14 24h2v1H14zM17 24h1v1H17zM20 24h4v1H20zM26 24h1v1H26zM30 24h2v1H30zM33 24h1v1H33zM36,24 h1v1H36zM3 25h2v1H3zM7 25h1v1H7zM11 25h1v1H11zM13 25h1v1H13zM17 25h1v1H17zM20 25h1v1H20zM22 25h2v1H22zM25 25h1v1H25zM30 25h2v1H30zM33 25h1v1H33zM36,25 h1v1H36zM0 26h1v1H0zM4 26h1v1H4zM6 26h4v1H6zM12 26h1v1H12zM14 26h3v1H14zM18 26h1v1H18zM21 26h2v1H21zM26 26h1v1H26zM28 26h1v1H28zM30 26h1v1H30zM32 26h1v1H32zM34 26h1v1H34zM36,26 h1v1H36zM1 27h4v1H1zM9 27h1v1H9zM12 27h1v1H12zM14 27h1v1H14zM18 27h2v1H18zM21 27h1v1H21zM23 27h1v1H23zM25 27h1v1H25zM27 27h1v1H27zM29 27h1v1H29zM31 27h1v1H31zM33 27h1v1H33zM35 27h1v1H35zM0 28h1v1H0zM2 28h1v1H2zM4 28h3v1H4zM8 28h1v1H8zM11 28h1v1H11zM20 28h1v1H20zM25 28h2v1H25zM28 28h6v1H28zM36,28 h1v1H36zM8 29h3v1H8zM12 29h6v1H12zM19 29h3v1H19zM25 29h2v1H25zM28 29h1v1H28zM32 29h3v1H32zM36,29 h1v1H36zM0 30h7v1H0zM9 30h5v1H9zM17 30h1v1H17zM19 30h1v1H19zM22 30h1v1H22zM24 30h1v1H24zM26 30h3v1H26zM30 30h1v1H30zM32 30h1v1H32zM34 30h1v1H34zM36,30 h1v1H36zM0 31h1v1H0zM6 31h1v1H6zM8 31h1v1H8zM10 31h4v1H10zM15 31h1v1H15zM17 31h3v1H17zM21 31h1v1H21zM23 31h1v1H23zM25 31h1v1H25zM27 31h2v1H27zM32 31h2v1H32zM35 31h1v1H35zM0 32h1v1H0zM2 32h3v1H2zM6 32h1v1H6zM8 32h1v1H8zM11 32h2v1H11zM18 32h1v1H18zM21 32h1v1H21zM24 32h3v1H24zM28 32h6v1H28zM36,32 h1v1H36zM0 33h1v1H0zM2 33h3v1H2zM6 33h1v1H6zM9 33h2v1H9zM12 33h1v1H12zM15 33h1v1H15zM18 33h2v1H18zM22 33h6v1H22zM29 33h1v1H29zM32 33h2v1H32zM35,33 h2v1H35zM0 34h1v1H0zM2 34h3v1H2zM6 34h1v1H6zM8 34h1v1H8zM11 34h6v1H11zM22 34h3v1H22zM26 34h1v1H26zM28 34h2v1H28zM32 34h1v1H32zM34 34h1v1H34zM36,34 h1v1H36zM0 35h1v1H0zM6 35h1v1H6zM10 35h2v1H10zM14 35h6v1H14zM24 35h2v1H24zM28 35h2v1H28zM31 35h1v1H31zM33 35h1v1H33zM35 35h1v1H35zM0 36h7v1H0zM9 36h1v1H9zM12 36h1v1H12zM15 36h1v1H15zM18 36h1v1H18zM20 36h4v1H20zM25 36h2v1H25zM28 36h1v1H28zM30 36h1v1H30zM32 36h2v1H32zM35,36 h2v1H35z"
          fill="#fff"
        />
      </svg>
    </div>
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-72o1r1-MuiTypography-root"
    >
      Send only 
      ASSET
       on 
      <span
        style="font-weight: 600;"
      >
        Polygon (Mumbai)
      </span>
       network to this deposit address. Sending any other coins may result in permanent loss.
    </h3>
    <div
      class="css-17iwtuv-MuiStack-root"
    >
      <div
        class="MuiBox-root css-p5x5a2"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-fsl47-MuiTypography-root"
        >
          Network
        </h4>
        <div
          class="MuiBox-root css-axw7ok"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1h1nqhj-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="networkIcon.png"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17h2jvq-MuiTypography-root"
          >
            Polygon (Mumbai)
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-p5x5a2"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-fsl47-MuiTypography-root"
        >
          Wallet Address
        </h4>
        <div
          class="MuiBox-root css-2jevur"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-y09ry9-MuiTypography-root"
          >
            0x0000000000000000000000000000000000000000
          </h2>
          <button
            aria-label="Copied!"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-d9f54s-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            data-testid="copy-button"
            tabindex="0"
            type="button"
          >
            <svg
              data-testid="ContentCopyIcon"
              fill="none"
              height="16px"
              viewBox="0 0 16 17"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.3333 6.5H7.33333C6.59695 6.5 6 7.09695 6 7.83333V13.8333C6 14.5697 6.59695 15.1667 7.33333 15.1667H13.3333C14.0697 15.1667 14.6667 14.5697 14.6667 13.8333V7.83333C14.6667 7.09695 14.0697 6.5 13.3333 6.5Z"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M3.33334 10.4997H2.66668C2.31305 10.4997 1.97392 10.3592 1.72387 10.1092C1.47382 9.8591 1.33334 9.51996 1.33334 9.16634V3.16634C1.33334 2.81272 1.47382 2.47358 1.72387 2.22353C1.97392 1.97348 2.31305 1.83301 2.66668 1.83301H8.66668C9.0203 1.83301 9.35944 1.97348 9.60949 2.22353C9.85953 2.47358 10 2.81272 10 3.16634V3.83301"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-4zkq9f"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-mox600-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`matches snapshot when variant is ReceiveVariant.Token 1`] = `
<div>
  <div
    class="MuiBox-root css-tfe0o3"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1rgr0y-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xtuuzo-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-16vochq-MuiTypography-root"
    >
      Receive ASSET
    </h2>
    <div
      class="MuiBox-root css-8h2w1h"
    />
  </div>
  <div
    class="MuiBox-root css-dodud4"
  >
    <div
      class="MuiBox-root css-1010uug"
    >
      <svg
        height="180"
        shape-rendering="crispEdges"
        viewBox="0 0 37 37"
        width="180"
      >
        <path
          d="M0,0 h37v37H0z"
          fill="#111619"
        />
        <path
          d="M0 0h7v1H0zM8 0h2v1H8zM13 0h4v1H13zM18 0h2v1H18zM21 0h1v1H21zM27 0h2v1H27zM30,0 h7v1H30zM0 1h1v1H0zM6 1h1v1H6zM10 1h6v1H10zM17 1h4v1H17zM22 1h4v1H22zM27 1h1v1H27zM30 1h1v1H30zM36,1 h1v1H36zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM10 2h3v1H10zM14 2h1v1H14zM16 2h1v1H16zM22 2h1v1H22zM24 2h1v1H24zM26 2h1v1H26zM28 2h1v1H28zM30 2h1v1H30zM32 2h3v1H32zM36,2 h1v1H36zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM10 3h4v1H10zM16 3h2v1H16zM20 3h2v1H20zM27 3h2v1H27zM30 3h1v1H30zM32 3h3v1H32zM36,3 h1v1H36zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h1v1H9zM11 4h3v1H11zM15 4h4v1H15zM20 4h2v1H20zM24 4h1v1H24zM27 4h2v1H27zM30 4h1v1H30zM32 4h3v1H32zM36,4 h1v1H36zM0 5h1v1H0zM6 5h1v1H6zM9 5h2v1H9zM13 5h2v1H13zM18 5h2v1H18zM21 5h1v1H21zM23 5h1v1H23zM25 5h1v1H25zM27 5h1v1H27zM30 5h1v1H30zM36,5 h1v1H36zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22 6h1v1H22zM24 6h1v1H24zM26 6h1v1H26zM28 6h1v1H28zM30,6 h7v1H30zM9 7h4v1H9zM14 7h3v1H14zM22 7h2v1H22zM25 7h1v1H25zM27 7h1v1H27zM2 8h1v1H2zM4 8h3v1H4zM8 8h1v1H8zM10 8h1v1H10zM16 8h1v1H16zM18 8h1v1H18zM21 8h2v1H21zM24 8h3v1H24zM29 8h1v1H29zM33 8h1v1H33zM36,8 h1v1H36zM2 9h1v1H2zM7 9h2v1H7zM10 9h3v1H10zM14 9h1v1H14zM16 9h1v1H16zM19 9h1v1H19zM23 9h4v1H23zM30 9h2v1H30zM33 9h1v1H33zM36,9 h1v1H36zM0 10h3v1H0zM6 10h4v1H6zM13 10h1v1H13zM15 10h1v1H15zM19 10h1v1H19zM21 10h1v1H21zM23 10h2v1H23zM26 10h1v1H26zM28 10h1v1H28zM30 10h1v1H30zM32 10h1v1H32zM34 10h1v1H34zM36,10 h1v1H36zM1 11h2v1H1zM5 11h1v1H5zM11 11h4v1H11zM16 11h1v1H16zM18 11h1v1H18zM20 11h1v1H20zM24 11h2v1H24zM27 11h1v1H27zM29 11h1v1H29zM31 11h1v1H31zM33 11h1v1H33zM35 11h1v1H35zM4 12h7v1H4zM12 12h1v1H12zM15 12h2v1H15zM18 12h1v1H18zM20 12h1v1H20zM25 12h2v1H25zM30 12h2v1H30zM33 12h1v1H33zM36,12 h1v1H36zM0 13h1v1H0zM2 13h1v1H2zM5 13h1v1H5zM7 13h1v1H7zM9 13h4v1H9zM14 13h2v1H14zM22 13h5v1H22zM30 13h2v1H30zM33 13h1v1H33zM36,13 h1v1H36zM1 14h3v1H1zM6 14h2v1H6zM9 14h1v1H9zM11 14h2v1H11zM16 14h1v1H16zM19 14h1v1H19zM21 14h3v1H21zM26 14h1v1H26zM28 14h1v1H28zM30 14h1v1H30zM32 14h1v1H32zM34 14h1v1H34zM36,14 h1v1H36zM1 15h1v1H1zM3 15h2v1H3zM7 15h2v1H7zM14 15h1v1H14zM17 15h1v1H17zM19 15h1v1H19zM21 15h3v1H21zM25 15h1v1H25zM27 15h1v1H27zM29 15h1v1H29zM31 15h1v1H31zM35 15h1v1H35zM0 16h3v1H0zM6 16h1v1H6zM8 16h2v1H8zM13 16h7v1H13zM21 16h1v1H21zM23 16h1v1H23zM25 16h2v1H25zM30 16h2v1H30zM33 16h1v1H33zM36,16 h1v1H36zM0 17h3v1H0zM4 17h2v1H4zM9 17h1v1H9zM12 17h1v1H12zM16 17h1v1H16zM18 17h1v1H18zM21 17h6v1H21zM30 17h2v1H30zM33 17h1v1H33zM36,17 h1v1H36zM0 18h1v1H0zM4 18h4v1H4zM10 18h3v1H10zM14 18h1v1H14zM16 18h4v1H16zM21 18h1v1H21zM24 18h1v1H24zM26 18h1v1H26zM28 18h1v1H28zM30 18h1v1H30zM32 18h1v1H32zM34 18h1v1H34zM36,18 h1v1H36zM0 19h2v1H0zM3 19h2v1H3zM8 19h1v1H8zM12 19h1v1H12zM15 19h1v1H15zM23 19h3v1H23zM27 19h1v1H27zM29 19h1v1H29zM31 19h1v1H31zM33 19h1v1H33zM35,19 h2v1H35zM1 20h3v1H1zM5 20h2v1H5zM10 20h2v1H10zM14 20h1v1H14zM17 20h1v1H17zM20 20h1v1H20zM24 20h3v1H24zM30 20h2v1H30zM33 20h1v1H33zM2 21h1v1H2zM8 21h2v1H8zM11 21h1v1H11zM16 21h5v1H16zM22 21h1v1H22zM25 21h2v1H25zM30 21h2v1H30zM33 21h1v1H33zM36,21 h1v1H36zM0 22h2v1H0zM3 22h1v1H3zM5 22h5v1H5zM11 22h1v1H11zM13 22h1v1H13zM16 22h1v1H16zM18 22h4v1H18zM23 22h2v1H23zM26 22h1v1H26zM28 22h1v1H28zM30 22h1v1H30zM32 22h1v1H32zM34 22h1v1H34zM36,22 h1v1H36zM0 23h4v1H0zM7 23h2v1H7zM12 23h2v1H12zM16 23h2v1H16zM20 23h1v1H20zM24 23h2v1H24zM27 23h1v1H27zM29 23h1v1H29zM31 23h1v1H31zM33 23h1v1H33zM35 23h1v1H35zM0 24h2v1H0zM4 24h6v1H4zM11 24h1v1H11zM14 24h2v1H14zM17 24h1v1H17zM20 24h4v1H20zM26 24h1v1H26zM30 24h2v1H30zM33 24h1v1H33zM36,24 h1v1H36zM3 25h2v1H3zM7 25h1v1H7zM11 25h1v1H11zM13 25h1v1H13zM17 25h1v1H17zM20 25h1v1H20zM22 25h2v1H22zM25 25h1v1H25zM30 25h2v1H30zM33 25h1v1H33zM36,25 h1v1H36zM0 26h1v1H0zM4 26h1v1H4zM6 26h4v1H6zM12 26h1v1H12zM14 26h3v1H14zM18 26h1v1H18zM21 26h2v1H21zM26 26h1v1H26zM28 26h1v1H28zM30 26h1v1H30zM32 26h1v1H32zM34 26h1v1H34zM36,26 h1v1H36zM1 27h4v1H1zM9 27h1v1H9zM12 27h1v1H12zM14 27h1v1H14zM18 27h2v1H18zM21 27h1v1H21zM23 27h1v1H23zM25 27h1v1H25zM27 27h1v1H27zM29 27h1v1H29zM31 27h1v1H31zM33 27h1v1H33zM35 27h1v1H35zM0 28h1v1H0zM2 28h1v1H2zM4 28h3v1H4zM8 28h1v1H8zM11 28h1v1H11zM20 28h1v1H20zM25 28h2v1H25zM28 28h6v1H28zM36,28 h1v1H36zM8 29h3v1H8zM12 29h6v1H12zM19 29h3v1H19zM25 29h2v1H25zM28 29h1v1H28zM32 29h3v1H32zM36,29 h1v1H36zM0 30h7v1H0zM9 30h5v1H9zM17 30h1v1H17zM19 30h1v1H19zM22 30h1v1H22zM24 30h1v1H24zM26 30h3v1H26zM30 30h1v1H30zM32 30h1v1H32zM34 30h1v1H34zM36,30 h1v1H36zM0 31h1v1H0zM6 31h1v1H6zM8 31h1v1H8zM10 31h4v1H10zM15 31h1v1H15zM17 31h3v1H17zM21 31h1v1H21zM23 31h1v1H23zM25 31h1v1H25zM27 31h2v1H27zM32 31h2v1H32zM35 31h1v1H35zM0 32h1v1H0zM2 32h3v1H2zM6 32h1v1H6zM8 32h1v1H8zM11 32h2v1H11zM18 32h1v1H18zM21 32h1v1H21zM24 32h3v1H24zM28 32h6v1H28zM36,32 h1v1H36zM0 33h1v1H0zM2 33h3v1H2zM6 33h1v1H6zM9 33h2v1H9zM12 33h1v1H12zM15 33h1v1H15zM18 33h2v1H18zM22 33h6v1H22zM29 33h1v1H29zM32 33h2v1H32zM35,33 h2v1H35zM0 34h1v1H0zM2 34h3v1H2zM6 34h1v1H6zM8 34h1v1H8zM11 34h6v1H11zM22 34h3v1H22zM26 34h1v1H26zM28 34h2v1H28zM32 34h1v1H32zM34 34h1v1H34zM36,34 h1v1H36zM0 35h1v1H0zM6 35h1v1H6zM10 35h2v1H10zM14 35h6v1H14zM24 35h2v1H24zM28 35h2v1H28zM31 35h1v1H31zM33 35h1v1H33zM35 35h1v1H35zM0 36h7v1H0zM9 36h1v1H9zM12 36h1v1H12zM15 36h1v1H15zM18 36h1v1H18zM20 36h4v1H20zM25 36h2v1H25zM28 36h1v1H28zM30 36h1v1H30zM32 36h2v1H32zM35,36 h2v1H35z"
          fill="#fff"
        />
        <image
          height="4.933333333333334"
          preserveAspectRatio="none"
          width="4.933333333333334"
          x="16.03333333333333"
          xlink:href="icon.png"
          y="16.03333333333333"
        />
      </svg>
    </div>
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-72o1r1-MuiTypography-root"
    >
      Send only 
      ASSET
       on 
      <span
        style="font-weight: 600;"
      >
        Polygon (Mumbai)
      </span>
       network to this deposit address. Sending any other coins may result in permanent loss.
    </h3>
    <div
      class="css-17iwtuv-MuiStack-root"
    >
      <div
        class="MuiBox-root css-p5x5a2"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-fsl47-MuiTypography-root"
        >
          Network
        </h4>
        <div
          class="MuiBox-root css-axw7ok"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1h1nqhj-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="networkIcon.png"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17h2jvq-MuiTypography-root"
          >
            Polygon (Mumbai)
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-p5x5a2"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-fsl47-MuiTypography-root"
        >
          Wallet Address
        </h4>
        <div
          class="MuiBox-root css-2jevur"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-y09ry9-MuiTypography-root"
          >
            0x0000000000000000000000000000000000000000
          </h2>
          <button
            aria-label="Copied!"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-d9f54s-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            data-testid="copy-button"
            tabindex="0"
            type="button"
          >
            <svg
              data-testid="ContentCopyIcon"
              fill="none"
              height="16px"
              viewBox="0 0 16 17"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.3333 6.5H7.33333C6.59695 6.5 6 7.09695 6 7.83333V13.8333C6 14.5697 6.59695 15.1667 7.33333 15.1667H13.3333C14.0697 15.1667 14.6667 14.5697 14.6667 13.8333V7.83333C14.6667 7.09695 14.0697 6.5 13.3333 6.5Z"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M3.33334 10.4997H2.66668C2.31305 10.4997 1.97392 10.3592 1.72387 10.1092C1.47382 9.8591 1.33334 9.51996 1.33334 9.16634V3.16634C1.33334 2.81272 1.47382 2.47358 1.72387 2.22353C1.97392 1.97348 2.31305 1.83301 2.66668 1.83301H8.66668C9.0203 1.83301 9.35944 1.97348 9.60949 2.22353C9.85953 2.47358 10 2.81272 10 3.16634V3.83301"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-p5x5a2"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-fsl47-MuiTypography-root"
        >
          Balance
        </h4>
        <div
          class="MuiBox-root css-69i1ev"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-bvqlkz-MuiTypography-root"
            data-testid="amount-display"
          >
            10 ASSET
          </h2>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5h7vtp-MuiTypography-root"
            data-testid="currency-display"
          >
            $10.00
          </h4>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-4zkq9f"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-mox600-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
